tfx/orchestration/experimental/kubernetes/kubernetes_dag_runner.py

```diff
@@ -0,0 +1,378 @@
+  Can be pacthed for testing purpose.
+  """
+  return kube_utils.is_inside_cluster()
+
+
+def get_default_kubernetes_metadata_config(
+) -> metadata_store_pb2.ConnectionConfig:
+  """Returns the default metadata connection config for a kubernetes cluster.
+
+  Returns:
+    A config proto that will be serialized as JSON and passed to the running
+    container so the TFX component driver is able to communicate with MLMD in
+    a kubernetes cluster.
+  """
+  connection_config = metadata_store_pb2.ConnectionConfig()
+  connection_config.mysql.host = 'mysql'
+  connection_config.mysql.port = 3306
+  connection_config.mysql.database = 'mysql'
+  connection_config.mysql.user = 'root'
+  connection_config.mysql.password = ''
+  return connection_config
+
+
+def _wrap_container_component(
+    component: base_node.BaseNode,
+    component_launcher_class:
+        Type[base_component_launcher.BaseComponentLauncher],
+    component_config: Optional[base_component_config.BaseComponentConfig],
+    pipeline: tfx_pipeline.Pipeline,
+) -> base_node.BaseNode:
+  """Wrapper for container component.
+
+  Args:
+  component: Component to be executed.
+  component_launcher_class: The class of the launcher to launch the
+    component.
+  component_config: component config to launch the component.
+  pipeline: Logical pipeline that contains pipeline related information.
+  """
+
+  # Reference: tfx.orchestration.kubeflow.base_component
+  component_launcher_class_path = '.'.join([
+      component_launcher_class.__module__, component_launcher_class.__name__
+  ])
+
+  serialized_component = utils.replace_placeholder(
+      json_utils.dumps(node_wrapper.NodeWrapper(component)))
+
+  arguments = [
+      '--pipeline_name',
+      pipeline.pipeline_info.pipeline_name,
+      '--pipeline_root',
+      pipeline.pipeline_info.pipeline_root,
+      '--run_id',
+      pipeline.pipeline_info.run_id,
+      '--metadata_config',
+      json_format.MessageToJson(
+          message=get_default_kubernetes_metadata_config(),
+          preserving_proto_field_name=True),
+      '--beam_pipeline_args',
+      json.dumps(pipeline.beam_pipeline_args),
+      '--additional_pipeline_args',
+      json.dumps(pipeline.additional_pipeline_args),
+      '--component_launcher_class_path',
+      component_launcher_class_path,
+      '--serialized_component',
+      serialized_component,
+      '--component_config',
+      json_utils.dumps(component_config),
+  ]
+
+  # Outputs/Parameters fields are not used as they are contained in
+  # the serialized component. We add a suffix to the component id
+  # to avoid MLMD conflict when registering this component.
+  return container_component.create_container_component(
+      name=component.id + _WRAPPER_SUFFIX,
+      outputs={},
+      parameters={},
+      image=_TFX_IMAGE,
+      command=_CONTAINER_COMMAND + arguments
+  )()
+
+
+class LaunchAsContainerComponent():
```

Comment:
Should this have a parent class? What part makes it hard for us to use the kubernetes component launcher class directly?

Reply:
I intended to implement it in a similar style as BeamDagRunner, but since it only uses the kubernetes_component_launcher as component_launcher_class, I agree that a function would be more succinct.